DSharpPlus.CommandsNext/Entities/Command.cs

```diff
@@ -1,194 +1,204 @@
-        /// </summary>
-        public ICommandModule? Module { get; internal set; }
-
-        /// <summary>
-        /// Gets the custom attributes defined on this command.
-        /// </summary>
-        public IReadOnlyList<Attribute> CustomAttributes { get; internal set; } = Array.Empty<Attribute>();
-
-        internal Command() { }
-
-        /// <summary>
-        /// Executes this command with specified context.
-        /// </summary>
-        /// <param name="ctx">Context to execute the command in.</param>
-        /// <returns>Command's execution results.</returns>
-        public virtual async Task<CommandResult> ExecuteAsync(CommandContext ctx)
+        try
         {
-            CommandResult res = default;
-            try
+            foreach (CommandOverload? overload in this.Overloads.OrderByDescending(x => x.Priority))
             {
-                var executed = false;
-                foreach (var ovl in this.Overloads.OrderByDescending(x => x.Priority))
+                ctx.Overload = overload;
+
+                // Attempt to match the arguments to the overload
+                ArgumentBindingResult args = await CommandsNextUtilities.BindArgumentsAsync(ctx, ctx.Config.IgnoreExtraArguments);
+                if (!args.IsSuccessful)
                 {
-                    ctx.Overload = ovl;
-                    var args = await CommandsNextUtilities.BindArgumentsAsync(ctx, ctx.Config.IgnoreExtraArguments);
-
-                    if (!args.IsSuccessful)
-                        continue;
-
-                    ctx.RawArguments = args.Raw;
-
-                    var mdl = ovl._invocationTarget ?? this.Module?.GetInstance(ctx.Services);
-                    if (mdl is BaseCommandModule bcmBefore)
-                        await bcmBefore.BeforeExecutionAsync(ctx);
-
-                    args.Converted[0] = mdl;
-                    var ret = (Task)ovl._callable.DynamicInvoke(args.Converted);
-                    await ret;
-                    executed = true;
-                    res = new CommandResult
-                    {
-                        IsSuccessful = true,
-                        Context = ctx
-                    };
-
-                    if (mdl is BaseCommandModule bcmAfter)
-                        await bcmAfter.AfterExecutionAsync(ctx);
-                    break;
+                    continue;
                 }
 
-                if (!executed)
-                    throw new ArgumentException("Could not find a suitable overload for the command.");
-            }
-            catch (Exception ex)
-            {
-                res = new CommandResult
+                ctx.RawArguments = args.Raw;
+
+                // From... what I can gather, this seems to be support for executing commands that don't inherit from BaseCommandModule.
+                // But, that can never be the case since all Commands must inherit from BaseCommandModule.
```

Comment:
note here, cnext ~technically~ supports dynamic registration of commands through CommandBuilder, in which case there's no inheritance from BCM